controllers/csi/gc/controller.go

```diff
@@ -0,0 +1,91 @@
+package csigc
+
+import (
+	"context"
+	"fmt"
+
+	dynatracev1alpha1 "github.com/Dynatrace/dynatrace-operator/api/v1alpha1"
+	dtcsi "github.com/Dynatrace/dynatrace-operator/controllers/csi"
+	"github.com/Dynatrace/dynatrace-operator/controllers/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/controllers/utils"
+	"github.com/Dynatrace/dynatrace-operator/dtclient"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// CSIGarbageCollector removes unused and outdated agent versions
+type CSIGarbageCollector struct {
+	client       client.Client
+	logger       logr.Logger
+	opts         dtcsi.CSIOptions
+	dtcBuildFunc dynakube.DynatraceClientFunc
+}
+
+// NewReconciler returns a new CSIGarbageCollector
+func NewReconciler(client client.Client, opts dtcsi.CSIOptions) *CSIGarbageCollector {
+	return &CSIGarbageCollector{
+		client:       client,
+		logger:       log.Log.WithName("csi.gc.controller"),
+		opts:         opts,
+		dtcBuildFunc: dynakube.BuildDynatraceClient,
+	}
+}
+
+func (r *CSIGarbageCollector) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&dynatracev1alpha1.DynaKube{}).
+		Complete(r)
+}
+
+var _ reconcile.Reconciler = &CSIGarbageCollector{}
+
+func (r *CSIGarbageCollector) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	r.logger.Info("reconciling csi driver", "namespace", request.Namespace, "name", request.Name)
```

Comment:
The message is misleading. It should be something like _running OneAgent garbage collection_.